src/bindings/c/include/openvino/c/ov_core.h

```diff
@@ -0,0 +1,268 @@
+
+/**
+ * @brief Release the memory allocated by ov_core_t.
+ * @ingroup Core
+ * @param core A pointer to the ov_core_t to free memory.
+ */
+OPENVINO_C_API(void) ov_core_free(ov_core_t* core);
+
+/**
+ * @brief Reads models from IR/ONNX/PDPD formats.
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_path Path to a model.
+ * @param bin_path Path to a data file.
+ * For IR format (*.bin):
+ *  * if path is empty, will try to read a bin file with the same name as xml and
+ *  * if the bin file with the same name is not found, will load IR without weights.
+ * For ONNX format (*.onnx):
+ *  * the bin_path parameter is not used.
+ * For PDPD format (*.pdmodel)
+ *  * the bin_path parameter is not used.
+ * @param model A pointer to the newly created model.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model(const ov_core_t* core, const char* model_path, const char* bin_path, ov_model_t** model);
+
+/**
+ * @brief Reads models from IR/ONNX/PDPD formats.
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_str String with a model in IR/ONNX/PDPD format.
+ * @param weights Shared pointer to a constant tensor with weights.
+ * @param model A pointer to the newly created model.
+ * Reading ONNX/PDPD models does not support loading weights from the @p weights tensors.
+ * @note Created model object shares the weights with the @p weights object.
+ * Thus, do not create @p weights on temporary data that can be freed later, since the model
+ * constant data will point to an invalid memory.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model_from_memory(const ov_core_t* core,
+                               const char* model_str,
+                               const ov_tensor_t* weights,
+                               ov_model_t** model);
+
+/**
+ * @brief Creates a compiled model from a source model object.
+ * Users can create as many compiled models as they need and use
+ * them simultaneously (up to the limitation of the hardware resources).
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model Model object acquired from Core::read_model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_compile_model(const ov_core_t* core,
+                      const ov_model_t* model,
+                      const char* device_name,
+                      ov_compiled_model_t** compiled_model,
+                      const ov_property_t* property);
```

Comment:
should not compiled_model be the last arg? Usually, in C output arguments are the last ones

Reply:
Actually, C API didn't apply the property in compile model, just keep it as a placeholder. Maybe we can put it before output arg.